controllers/front/notify.php

```diff
@@ -294,12 +286,39 @@ public function validateAmount()
                 $this->saveLog(array(
                     'message' => $this->amountMismatchError
                 ));
-            } catch (\Exception $e) {
+            } catch (\Exception $exception) {
                 // Do nothing
             }
         }
     }
 
+    /**
+     * Check that the merchant order was not previously processes and is ready to be paid
+     *
+     * @throws Exception
+     */
+    public function checkMerchantOrderStatus()
+    {
+        try {
+            if ($this->merchantOrder->orderExists() !== false) {
+                throw new WrongStatusException('PS->orderExists(): already_processed');
+            }
+
+            // Double check
+            $tableName = _DB_PREFIX_ . 'pagantis_order';
+            $sql = ('select ps_order_id from `' . $tableName . '` where `id` = ' . $this->merchantOrderId
+                . ' and `order_id` = \'' . $this->pagantisOrderId . '\''
+                . ' and `ps_order_id` is not null');
+            $results = Db::getInstance()->ExecuteS($sql);
+            if (is_array($results) && count($results) === 1) {
+                throw new WrongStatusException('PS->record found in ' . $tableName . ': already_processed');
```

Comment:
could be good write the order_id in the log information

Reply:
Good idea, done.